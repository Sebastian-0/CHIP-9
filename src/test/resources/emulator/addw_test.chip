# Test simple adds
ldi A, 1
ldx BC, 0xFFFF
ldx DE, 0x7FFF
ldx HL, 0xF

# Carry, Half-Carry and Zero flag
addx BC
sout FLAGS

# Half-Carry and Negative flag
addx DE
sout FLAGS

# No flag
addx HL
sout FLAGS

sout BC
sout DE
sout HL

# Zero flag
ldi A, 0
ldx BC, 0
addx BC

sout FLAGS
sout BC

hcf