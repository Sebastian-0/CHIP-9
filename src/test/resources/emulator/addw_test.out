176
96
0

0
32768
16

128
0